ister_test.py

```diff
@@ -3800,6 +3801,18 @@ def mock_modify_cloud_init_service_file(target_dir):
     commands_compare_helper(commands)
 
 
+def gui_network_connection_with_proxy():
+    netreq = ister_gui.NetworkRequirements()
+    netreq.config = {"proxy": "https://proxy.to.clearlinux.org:1080"}
+    netreq._network_connection()
```

Comment:
So I'd mock out the pycurl.Curl() to give you a class you just introspec to see that the netreq.config is applied correctly and to not actually perform the request.